otlp/traces_direct_json.go

```diff
@@ -0,0 +1,1115 @@
+	ctx context.Context,
+	v *fastjson.Value,
+	key []byte,
+	attrs *msgpAttributes,
+	depth int,
+) error {
+	obj, err := v.Object()
+	if err != nil {
+		return nil // Skip invalid AnyValue
+	}
+
+	// Visit the object once to find the value type
+	obj.Visit(func(k []byte, v *fastjson.Value) {
+		switch string(k) {
+		case "stringValue":
+			value := v.GetStringBytes()
+
+			// Special handling for sample rate
+			if bytes.Equal(key, []byte("sampleRate")) || bytes.Equal(key, []byte("SampleRate")) {
+				var f float64
+				f, err = strconv.ParseFloat(string(value), 64)
+				if err == nil {
+					if attrs.sampleRate == 0 || bytes.Equal(key, []byte("sampleRate")) {
+						attrs.sampleRate = sampleRateFromFloat(f)
+					}
+				}
+				return
+			}
+
+			// Check for service name
+			if bytes.Equal(key, []byte("service.name")) {
+				attrs.serviceName = string(value)
+			}
+
+			attrs.addString(key, value)
+
+		case "boolValue":
+			attrs.addBool(key, v.GetBool())
+
+		case "intValue":
+			// JSON encodes int64 as string
+			strBytes := v.GetStringBytes()
+			var val int64
+			val, err = strconv.ParseInt(string(strBytes), 10, 64)
+			if err != nil {
+				return
+			}
+
+			// Special handling for sample rate
+			if bytes.Equal(key, []byte("sampleRate")) || bytes.Equal(key, []byte("SampleRate")) {
+				if attrs.sampleRate == 0 || bytes.Equal(key, []byte("sampleRate")) {
+					attrs.sampleRate = int32(max(int64(defaultSampleRate), min(val, math.MaxInt32)))
+				}
+				return
+			}
+
+			attrs.addInt64(key, val)
+
+		case "doubleValue":
+			floatVal := v.GetFloat64()
+
+			// Special handling for sample rate
+			if bytes.Equal(key, []byte("sampleRate")) || bytes.Equal(key, []byte("SampleRate")) {
+				if attrs.sampleRate == 0 || bytes.Equal(key, []byte("sampleRate")) {
+					attrs.sampleRate = sampleRateFromFloat(floatVal)
+				}
+				return
+			}
+
+			attrs.addFloat64(key, floatVal)
+
+		case "bytesValue":
+			// Bytes are base64 encoded in JSON
+			strBytes := v.GetStringBytes()
+			var b []byte
+			b, err = base64.StdEncoding.DecodeString(string(strBytes))
+			if err != nil {
+				return
+			}
+
+			husky.AddTelemetryAttribute(ctx, "received_bytes_attr_type", true)
+			attrs.addAny(key, addAttributeToMapAsJsonDirect(b))
+
+		case "arrayValue":
+			// For arrays, we need to parse and JSON encode
+			husky.AddTelemetryAttribute(ctx, "received_array_attr_type", true)
+			var arr []any
+			arr, err = unmarshalArrayValueJSON(ctx, v)
+			if err != nil {
+				return
+			}
+			attrs.addAny(key, addAttributeToMapAsJsonDirect(arr))
+
+		case "kvlistValue":
+			// For kvlists, handle flattening
+			husky.AddTelemetryAttributes(ctx, map[string]any{
+				"received_kvlist_attr_type": true,
+				"kvlist_max_depth":          depth,
+			})
+
+			if depth < maxDepth {
+				// Flatten the kvlist
+				if err = unmarshalKvlistValueFlattenJSON(ctx, v, key, attrs, depth+1); err != nil {
+					return
+				}
+			} else {
+				// Max depth exceeded, JSON encode the whole thing
```

Comment:
Can we add a test that exercise this code path?

Reply:
Good catch, I'll add JSON to `TestUnmarshalTraceRequestDirect_NestedMapAttributes` which covers this.